python/demo/demo_navier_stokes.py

```diff
@@ -0,0 +1,419 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.13.6
+# ---
+
+# # A divergence conforming discontinuous Galerkin method for the Navier-Stokes equations
+# This demo illustrates how to implement a divergence conforming
+# discontinuous Galerkin method for the Navier-Stokes equations in
+# FEniCSx. The method conserves mass exactly and uses upwinding. The
+# formulation is based on a combination of "A fully divergence-free
+# finite element method for magnetohydrodynamic equations" by Hiptmair
+# et al., "A Note on Discontinuous Galerkin Divergence-free Solutions
+# of the Navier-Stokes Equations" by Cockburn et al, and "On the Divergence
+# Constraint in Mixed Finite Element Methods for Incompressible Flows" by
+# John et al.
+
+# ## Governing equations
+# We consider the incompressible Navier-Stokes equations in a domain
+# $\Omega \subset \mathbb{R}^d$, $d \in \{2, 3\}$, and time interval
+# $(0, \infty)$, given by
+# $$
+#     \partial_t u - \nu \Delta u + (u \cdot \nabla)u + \nabla p = f
```

Comment:
Does `\nu` actually get used in the discretisation?

Reply:
The discretisation uses the dimensionless form, where `\nu` is incorporated into the Reynolds number. I'll make this clearer